### PR TITLE
Add PushManager and Notification for Server Push 

### PR DESCRIPTION
**Pull request by JayChl**
- Add router dir and main.js for apis
- Add PushManager and send-push-notification into push dir
- Add body-parser, request npm package for push

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -10,6 +10,8 @@
     "express": "~4.14.0",
     "morgan": "~1.7.0",
     "forever": "~0.15.3",
-    "nodemon": "latest"
+    "nodemon": "latest",
+    "body-parser": "~1.16.0",
+    "request": "latest"
```
Comment (coconutperm) [inline server/package.json]: I think better that we use specific version. how about you?

File: server/router/main.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+module.exports = function(app) {
```
Comment (coconutperm) [inline server/router/main.js]: why is it main?

Comment (romandev) [inline server/router/main.js]: I think this file should be located in server/push.
Or, we can consider 'public' directory to provide rest APIs to upper layer(e.g. client layer).

File: server/router/main.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  app.post('/api/client', function(req,res) {
+    // Add new client
+    const auth = req.headers['authorization'];
+    const clientToken = auth.split('=')[1];
+    if (clientToken) {
+      console.log('Store client token : ' + clientToken);
+
+      const pushManager = require('../push/push-manager.js');
```
Comment (romandev) [inline server/router/main.js]: I'm not sure but we might have to declare to require module at top of file. (as per our guide?)
Also, I didn't like that but the |const pushManager| should be |var pushManager|. (as per our guide?)

File: server/router/main.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+module.exports = function(app) {
+  app.post('/api/client', function(req,res) {
```
Comment (romandev) [inline server/router/main.js]: should be |function(request, response) {|

File: server/router/main.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      console.log('Remove client token : ' + clientToken);
+      // Remove token from DB
+      pushManager.removeClient(clientToken);
+      res.sendStatus(200);
+    } else {
+      res.sendStatus(400);
+    }
+  });
+
+  app.get('/api/client', function(req, res) {
```
Comment (romandev) [inline server/router/main.js]: Please remove unnecessary lines.

File: server/router/main.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      pushManager.sendPushNotification(clientToken);
+      // StoreClient token to DB
+      res.sendStatus(200);
+
+    } else {
+      res.sendStatus(400); // bad request
+    }
+  });
+
+  app.delete('/api/client', function(req,res) {
```
Comment (romandev) [inline server/router/main.js]: ditto

File: server/push/send-push-notification.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+var serverKey = 'Your server key here, Never expose to client code';
```
Comment (romandev) [inline server/push/send-push-notification.js]: please add one blank line.

**Pull request closed without merging**